openwrt-openwisp-monitoring/files/lib/openwisp/monitoring_utils.lua

```diff
@@ -54,4 +56,13 @@ function utils.is_excluded(name)
   return name=='lo'
 end
 
+function utils.popen(command)
+  local command_file=io.popen(command)
+  local output_file=assert(io.tmpfile())
+  output_file:write(command_file:read("*a"))
+  command_file:close()
+  output_file:seek('set',0)
+  return output_file
```

Comment:
shouldn't the file be removed once the output is read? And are you sure using tmp files is really necessary?

Wouldn't it be enough to just do:

```lua
output = commmand_file:read("*a")
command_file:close()
return output
```

We weren't doing this before explicitly for every call.